sfcasts/ep1/github-service-part-2.md

```diff
@@ -1,103 +1,83 @@
-
-Let's go ahead and do this logger. We want to log as info and our message will be
-request Dino issues. And we're going to add a little extra cont context to this
-message. And by passing an array, which will contain the Dino's name. So Dino, uh,
-for the key and the name is dinosaur name at what we always know, which, uh, Dino we
-were trying to get, uh, an issue from. And then below that, let's go ahead and throw
-in the response status. Ah, we're going to throw in the response status that was
-given back from our, uh, HTTP request. So response get status code, and we're going
-to return, uh, move back to her terminal. Let's run this test one more time. Oh no,
-we got two errors now. So both of our GitHub service test test get health report
-returns, correct health status for Dino with data sets, sick Dino and a healthy Dino
-are both saying that we have an argument count error, two few arguments to function
-at GitHub service construct, zero pass. And exactly one were expected looking at our,
-uh, service here. That's because we're supposed to be passing in from our test a
-logger instance, and we are not doing that yet. So coming up next, we'll show you how
-to do that. And we'll fix this.
+# GitHub Service: Implementation
 
+Now that we have an idea of what we need the `GithubService` to do, let's add
+the logic inside that will fetch the issues from the `dino-park` repository
+using GitHub's API.
+
+## Add the client and make a request
+
+To *make* HTTP requests, at your terminal, install Symfony's HTTP Client with:
+
+```terminal
+composer require symfony/http-client
+```
+
+Inside of `GithubService`, instantiate an HTTP client with
+`$client = HttpClient::create()`. To make a request, call `$client->request()`.
+This needs 2 things. 1st: what HTTP method to use, like `GET` or `POST`. In this
+case, it should be `GET`. 2nd: the URL. Use
+`https://api.github.com/repos/SymfonyCasts/dino-park/issues` to fetch all the
+"issues" from the `dino-park` repository.
+
+## Parse the HTTP Response
+
+Ok, now what? Looking back at the `dino-park` repo, GitHub will return a JSON 
+response that contains the issues we see here. Each issue has a title with a 
+dino's name and if the issue has a label attached to it, we'll get that back too.
+So, set `$client->get()` to a new `$response` variable. Then, below, `foreach()`
+over `$response->toArray()` as an `$issue`. The cool thing about using Symfony's
+HTTP Client is that we don't have to bother transforming the JSON from GitHub 
+into an array - `toArray()` does that heavy lifting for us. Inside this loop,
+check if the issue title contains the `$dinosaurName`. So
+`if (str_contains($issue['title'], $dinosaurName))` then we'll `// Do Something`
+with that issue.
+
+At this point, we've found the issue for our dinosaur. Woo! Now we need to
+loop over each label to see if we can find the health status. To help, I'll
+paste in a private method: you can copy this from the code block on this page.
```

Comment:
Do we add in a "```" below for the pasted code?

Reply:
Good question, but no. After we finish publishing the video, someone on the SfCasts team goes back in and adds code to include the "magic code blocks" - e.g. https://symfonycasts.com/screencast/symfony-doctrine/entity#codeblock-33b9a4adde - you can see how that code block looks here - https://github.com/SymfonyCasts/symfony6/blob/main/sfcasts/ep3-doctrine/entity.md - it's got a format like `[[[ code('33b9a4adde') ]]]` - this is a key that we create internally that ultimately pulls the code for a given file, at a given TutsHero step, showing a certain sub-set of lines. But the point is, this is all added later.